File: src/lib/software/teamviewer.py
from src.lib.software import BaseSoftware

class TeamViewer(BaseSoftware):
    def __init__(self):
        super().__init__()

        self.download_name = 'TeamViewer_Setup_x64.exe'
        self.silent_install_args = ['/S']

    def resolve_download_url(self):
        self.url_resolved.emit('https://download.teamviewer.com/download/TeamViewer_Setup_x64.exe')

File: packages/app/src/services/software/definitions/teamviewer.ts
import { ok } from 'neverthrow';
import { SoftwareCategory } from 'shared';
import type { ISoftwareDefinition } from 'shared';

export default class implements ISoftwareDefinition {
	public key = 'teamviewer';
	public name = 'TeamViewer';
	public category = [SoftwareCategory.Social, SoftwareCategory.Utility];
	public downloadName = 'TeamViewer_Setup_x64.exe';
	public icon = 'teamviewer.png';
	public homepage = 'https://teamviewer.com';

	public async resolveDownloadUrl() {
		return ok('https://download.teamviewer.com/download/TeamViewer_Setup_x64.exe');
	}
}
